Exercise_1.java

```diff
@@ -1,8 +1,25 @@
 class BinarySearch { 
     // Returns index of x if it is present in arr[l.. r], else return -1 
-    int binarySearch(int arr[], int l, int r, int x) 
+    int binarySearch(int[] arr, int l, int r, int x) 
     { 
         //Write your code here
+        if(l<=r){

```

Comment:
Good Job! Please add time and space complexity and comments explaining your approach.